plugins/pip/venvShellHook.sh

```diff
@@ -1,8 +1,54 @@
 #!/bin/sh
+set -eu
+STATE_FILE="$DEVBOX_PROJECT_ROOT/.devbox/venv_check_completed"
 
-if ! [ -d "$VENV_DIR" ]; then
-    echo "Creating new venv environment in path: '${VENV_DIR}'"
-    python3 -m venv "$VENV_DIR"
-    echo "You can activate the virtual environment by running '. \$VENV_DIR/bin/activate' (for fish shell, replace '.' with 'source')" >&2
+is_valid_venv() {
+    [ -f "$1/bin/activate" ] && [ -f "$1/bin/python" ]
+}
+
+is_devbox_venv() {
+    [ "$1/bin/python" -ef "$DEVBOX_PACKAGES_DIR/bin/python" ]
+}
+
+create_venv() {
+    python -m venv "$VENV_DIR" --clear
+    echo "*\n.*" >> "$VENV_DIR/.gitignore"
+}
+
+# Check if we've already run this script
+if [ -f "$STATE_FILE" ]; then
+    # "We've already run this script. Exiting..."
+    exit 0
 fi
 
+# Check that Python version supports venv
+if ! python -c 'import venv' 1> /dev/null 2> /dev/null; then
+    echo "\033[1;33mWARNING: Python version must be > 3.3 to create a virtual environment.\033[0m"
+    touch "$STATE_FILE"
+    exit 1
```

Comment:
Yes it should, because the python in the venv is a symlink to the python in our Devbox profile, which is a symlink to the nix store. I'll test a few times to make sure

Reply:
Confirmed, once the `.venv` is created, updating python automatically updates the python linked in the `.venv` folder